From 92bbcd78fda003c8f37ed8db40a0185406184277 Mon Sep 17 00:00:00 2001
From: Alper Nebi Yasak <alpernebiyasak@gmail.com>
Date: Wed, 10 Jul 2024 14:32:19 +0300
Subject: [PATCH 4/6] HACK: rk3399: gru: Remove assigned clock dt properties
 for EDP node

Having the PCLK_EDP clock in the assigned-clocks property of the `edp`
node means that U-Boot tries to set its rate automatically. This clock
isn't implemented for the RK3399 clock driver, so it fails and prevents
display from being initialized.

The display happens to work fine without it, remove the property until
the clock driver can handle the clock.

Signed-off-by: Alper Nebi Yasak <alpernebiyasak@gmail.com>
---
 arch/arm/dts/rk3399-gru-u-boot.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/dts/rk3399-gru-u-boot.dtsi b/arch/arm/dts/rk3399-gru-u-boot.dtsi
index dfc7be4c621..e563a607864 100644
--- a/arch/arm/dts/rk3399-gru-u-boot.dtsi
+++ b/arch/arm/dts/rk3399-gru-u-boot.dtsi
@@ -29,6 +29,9 @@
 
 &edp {
 	rockchip,panel = <&edp_panel>;
+
+	/delete-property/ assigned-clocks;
+	/delete-property/ assigned-clock-rates;
 };
 
 &emmc_phy {
-- 
2.47.3

